pkg/metrics/backends/prometheus/prometheus.go

```diff
@@ -93,3 +97,51 @@ func (b *Backend) QueryInstant(ctx context.Context, query string, ts time.Time)
 		return nil, errors.Errorf("unknown supported type: '%q'", v)
 	}
 }
+
+// getAlertmanagers returns Alertmanager clients that is associated with the backend.
+func (b *Backend) getAlertmanagers(ctx context.Context) ([]*amclient.Alertmanager, error) {
+	level.Debug(b.logger).Log("msg", "Get active Alertmanagers")
+	amResult, err := b.prometheus.AlertManagers(ctx)
```

Comment:
what if there is no Alertmanager configured?

Reply:
It's fine, amResult will be an empty slice.